### PR TITLE
Sparkling ExampleHerb code

### PR DESCRIPTION
**Pull request by GoodPro712**
### What are the changes to ExampleMod?
Cleaned up ExampleHerb quite a bit. The original code looked really ugly so I tried to make it better

### CODE REVIEW

File: ExampleMod/Tiles/ExampleHerb.cs [hunk 1]
```diff
@@ -7,78 +9,82 @@
 
 namespace ExampleMod.Tiles
 {
+	//A plant with 3 stages, planted, growing and grown.
+	//Sadly, modded plants are unable to be grown by the flower boots
 	public class ExampleHerb : ModTile
 	{
-		public override void SetDefaults() {
+		private static readonly byte pixelsPerStage = 18; //a field for readibilty and to kick out those magic numbers
```
Comment (Jofairden) [inline ExampleMod/Tiles/ExampleHerb.cs:16]: Just make this `const`

File: ExampleMod/Tiles/ExampleHerb.cs [hunk 2]
```diff
@@ -7,78 +9,82 @@
...
+
+		//A method to quickly get the current stage of the herb
+		private Stage GetStage(int i, int j)
+		{
+			Tile tile = Framing.GetTileSafely(i, j); //Always use Framing.GetTileSafely instead of Main.tile as it prevents any errors caused from other mods
+			return (Stage)(tile.frameX / pixelsPerStage);
+		}
+
+		//An enum on the 3 stages of herb growth.
+		private enum Stage : byte
```
Comment (Jofairden) [inline ExampleMod/Tiles/ExampleHerb.cs:83]: This should be above the const, preferably even outside the class as `PlantStage` when used for multiple plants.

File: ExampleMod/Tiles/ExampleHerb.cs (hunk 1 above)
Comment (Jofairden) [reply ExampleMod/Tiles/ExampleHerb.cs:16]: Also name it FrameHeight instead

File: ExampleMod/Tiles/ExampleHerb.cs [hunk 3]
```diff
@@ -6,79 +8,83 @@
...
+		Growing,
+		Grown
+	}
+
+	//A plant with 3 stages, planted, growing and grown.
+	//Sadly, modded plants are unable to be grown by the flower boots
 	public class ExampleHerb : ModTile
 	{
-		public override void SetDefaults() {
+		private const byte FrameWidth = 18; //a field for readibilty and to kick out those magic numbers
```
Comment (Jofairden) [inline ExampleMod/Tiles/ExampleHerb.cs:24]: Should be int, but alright

Comment (GoodPro712) [reply ExampleMod/Tiles/ExampleHerb.cs:24]: What's your reason why?

Comment (Jofairden) [reply ExampleMod/Tiles/ExampleHerb.cs:24]: Because it's just a number, the height of a frame. It has nothing to do with bytes.

Comment (GoodPro712) [reply ExampleMod/Tiles/ExampleHerb.cs:24]: as far as I know it being a byte would reduce size

Comment (Jofairden) [reply ExampleMod/Tiles/ExampleHerb.cs:24]: You're not dealing with any networking or saving data here, so that doesn't matter.

**Pull request merged**